### ISSUE TITLE
[bug] MSBuild Helper converts target string to list implicitly

### ISSUE THREAD

**Issue by KerstinKeller** [conan-io/conan#6553]
When trying to build python with the `MSBuild` Build helper like this
```
msbuild = MSBuild(self)
msbuild.build(self.extracted_dir + "/PCbuild/pcbuild.sln", targets="pythoncore", upgrade_project=False, build_type="Release)   
``` 
It translates to the following build call for targets:
```
/target:p;y;t;h;o;n;c;o;r;e
```

Instead it seems that the correct call needs to be:
```
msbuild = MSBuild(self)
msbuild.build(self.extracted_dir + "/PCbuild/pcbuild.sln", targets=["pythoncore"], upgrade_project=False, build_type="Release)   
``` 

It would be very useful if Conan didn't implicitly split the targets argument to a list of chars, but complains that targets need to be a list instead. Or (more obviously) state that the Python type should be a list as argument type in the documentation.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version: Visual Studio 2019
  * Conan version: 1.22.2
  * Python version: 3.8

**Comment by memsharded** [comment]
Hi @KerstinKeller 

According to the docs:

> targets (Optional, Defaulted to None): Sets /target flag to the specified list of targets to build.

Yes, agree, this is not a bug per-se, but the best would be to raise an error if it is not a list, rather than splitting it, that doesn't make sense. Better error than the docs, IMHO.

Thanks for the suggestion, let's fix it for 1.23.

**Comment by memsharded** [comment]
Do you think https://github.com/conan-io/conan/pull/6555 will be good?

**Comment by KerstinKeller** [comment]
Yes perfect, looks good to me 👍

**Comment by memsharded** [comment]
Great! https://github.com/conan-io/conan/pull/6555 merged, will be released in 1.23
